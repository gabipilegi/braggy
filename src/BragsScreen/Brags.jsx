import React from 'react'
import { List, ListItem, ListItemIcon, Container } from '@material-ui/core'
import { ChatBubbleRounded } from '@material-ui/icons'
import { makeStyles } from '@material-ui/core/styles'

const Brags = ({ brags }) => {
  const useStyles = makeStyles((theme) => ({
    brag: {
      wordBreak: 'break-all',
    },
  }))

  const classes = useStyles()

  return (
    <Container>
      <List>
        {brags.map((brag, index) => (
          <div class={classes.brag}>
            <ListItem key={index}>
              <ListItemIcon>
                <ChatBubbleRounded />
              </ListItemIcon>
              {brag}
            </ListItem>
          </div>
        ))}
      </List>
    </Container>
  )
}

export default Brags
