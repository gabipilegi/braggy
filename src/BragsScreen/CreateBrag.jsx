import React from 'react'

const CreateBrag = ({ brag, onSubmit, onChange }) => {
  return (
    <form onSubmit={onSubmit} data-testid="brag-form">
      <label htmlFor="brag">Brag:</label>
      <input
        id="brag"
        type="text"
        placeholder="Enter Brag Here"
        value={brag}
        onChange={onChange}
      />
      <button className="button">Submit</button>
    </form>
  )
}

export default CreateBrag
