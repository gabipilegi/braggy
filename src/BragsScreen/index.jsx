import React, { useState } from 'react'
import Brags from './Brags'
import CreateBrag from './CreateBrag'
import { Grid } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import moment from 'moment'

const BragsScreen = () => {
  const [{ brag, brags }, setState] = useState({
    brag: { description: '' },
    brags: [],
  })

  const useStyles = makeStyles((theme) => ({
    title: {
      color: theme.palette.text.secondary,
    },
  }))

  const classes = useStyles()

  const onSubmit = (event) => {
    event.preventDefault()
    let newBrag = {
      description: brag.description,
      date: moment().format('L'),
    }
    setState({
      brag: { description: '' },
      brags: [...brags, newBrag],
    })
  }

  const onChange = (event) => {
    let value = event.target.value

    setState((previousState) => ({
      brag: { description: value },
      brags: previousState.brags,
    }))
  }

  return (
    <Grid container>
      <Grid item xs={12}>
        <p className={classes.title}>What are we gonna brag about today?</p>
        <CreateBrag brag={brag} onSubmit={onSubmit} onChange={onChange} />
      </Grid>
      <Grid item xs={12}>
        <Brags brags={brags} />
      </Grid>
    </Grid>
  )
}

export default BragsScreen
