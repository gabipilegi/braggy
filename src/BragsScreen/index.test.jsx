import React from 'react'
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import BragsScreen from '.'

describe('BragsScreen', () => {
  test('Should match snapshot', () => {
    const { container } = render(<BragsScreen />)
    expect(container).toMatchSnapshot()
  })

  test('Should add brag to list on submit', async () => {
    render(<BragsScreen />)
    const newBrag = 'Made a carrot cake'
    const input = screen.getByLabelText(/brag/i)
    const form = screen.getByTestId('brag-form')

    fireEvent.change(input, { target: { value: newBrag } })
    await waitFor(() => input)

    fireEvent.submit(form)
    await waitFor(() =>
      expect(screen.getByRole('listitem')).toHaveTextContent(newBrag)
    )
  })
})
