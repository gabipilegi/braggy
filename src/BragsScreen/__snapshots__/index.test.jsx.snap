// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BragsScreen Should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p
        class="makeStyles-title-1"
      >
        What are we gonna brag about today?
      </p>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <form
          data-testid="brag-form"
        >
          <div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
                for="brag"
                id="brag-label"
              >
                Your brag here
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
              >
                <textarea
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                  id="brag"
                  required=""
                  rows="4"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-4"
                  >
                    <span>
                      Your brag here
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-fullWidth"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Brag
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3.4 20.4l17.45-7.48c.81-.35.81-1.49 0-1.84L3.4 3.6c-.66-.29-1.39.2-1.39.91L2 9.12c0 .5.37.93.87.99L17 12 2.87 13.88c-.5.07-.87.5-.87 1l.01 4.61c0 .71.73 1.2 1.39.91z"
                    />
                  </svg>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <ul
          class="MuiList-root MuiList-padding"
        />
      </div>
    </div>
  </div>
</div>
`;
